libafl_qemu/src/injections.rs

```diff
@@ -0,0 +1,312 @@
+            };
+            let second_parameter = unsafe {
+                if (*c_array.offset(2)).is_null() {
+                    return SyscallHookResult::new(None);
+                }
+                CStr::from_ptr(*c_array.offset(2)).to_string_lossy()
+            };
+            if first_parameter == "-c"
+                && (second_parameter.to_lowercase().contains("';fuzz;'")
+                    || second_parameter.to_lowercase().contains("\";fuzz;\""))
+            {
+                panic!("Found command injection!");
+            }
+
+            //println!("PARAMETERS First {} Second {}", first_parameter, second_
+        }
+        SyscallHookResult::new(Some(0))
+    } else {
+        SyscallHookResult::new(None)
+    }
+}
+
+fn find_function(
+    emu: &Emulator,
+    file: &String,
+    function: &str,
+    loadaddr: GuestAddr,
+) -> Result<GuestAddr, Error> {
+    let mut elf_buffer = Vec::new();
+    let elf = EasyElf::from_file(file, &mut elf_buffer)?;
+    let offset = if loadaddr > 0 {
+        loadaddr
+    } else {
+        emu.load_addr()
+    };
+    let start_pc = elf
+        .resolve_symbol(function, offset)
+        .ok_or_else(|| Error::empty_optional("Symbol {function} not found in {file}"))?;
+    println!("Found {function} in {file}");
+    Ok(start_pc)
+}
+
+extern "C" fn on_call_check(val: u64, _pc: GuestAddr) {
+    let emu = Emulator::new_empty();
+    let parameter: u8 = (val & 0xff) as u8;
+    let off: usize = (val >> 8) as usize;
+
+    //println!("on_call_check {} {}", parameter, off);
+
+    #[cfg(any(target_arch = "x86_64", not(target_arch)))]
+    let reg: GuestAddr = match parameter {
+        0 => emu.current_cpu().unwrap().read_reg(Regs::Rdi).unwrap_or(0),
+        1 => emu.current_cpu().unwrap().read_reg(Regs::Rsi).unwrap_or(0),
+        2 => emu.current_cpu().unwrap().read_reg(Regs::Rdx).unwrap_or(0),
+        3 => emu.current_cpu().unwrap().read_reg(Regs::Rcx).unwrap_or(0),
+        4 => emu.current_cpu().unwrap().read_reg(Regs::R8).unwrap_or(0),
+        5 => emu.current_cpu().unwrap().read_reg(Regs::R9).unwrap_or(0),
+        _ => panic!("unknown register"),
+    };
+    #[cfg(cpu_target = "aarch64")]
+    let reg: GuestAddr = match parameter {
+        0 => emu.current_cpu().unwrap().read_reg(Regs::X0).unwrap_or(0),
+        1 => emu.current_cpu().unwrap().read_reg(Regs::X1).unwrap_or(0),
+        2 => emu.current_cpu().unwrap().read_reg(Regs::X2).unwrap_or(0),
+        3 => emu.current_cpu().unwrap().read_reg(Regs::X3).unwrap_or(0),
+        4 => emu.current_cpu().unwrap().read_reg(Regs::X4).unwrap_or(0),
+        5 => emu.current_cpu().unwrap().read_reg(Regs::X5).unwrap_or(0),
+        _ => panic!("unknown register"),
+    };
+
+    //println!("reg value = {:x}", reg);
+    if reg > 0 {
+        let query = unsafe {
+            let c_str_ptr = reg as *const c_char;
+            let c_str = CStr::from_ptr(c_str_ptr);
+            c_str.to_string_lossy()
```

Comment:
I don't really like that we are allocating a String for every call, we could instead compare bytes direclty

Reply:
how?

Reply:
I can try to throw something together

Reply:
yes please because I lack the rust skills :)

Reply:
Will do a new PR for that